static/app/utils/useRefChildrenVisibility.tsx

```diff
@@ -0,0 +1,60 @@
+import {useEffect, useState} from 'react';
+
+interface UseRefChildrenVisibilityOptions {
+  ref: React.MutableRefObject<HTMLElement | null>;
+  visibleRatio: number;
+}
+
+/**
+ * Determine the visibility of children elements within a scroll container.
+ */
+export function useRefChildrenVisibility({
+  ref,
+  visibleRatio,
+}: UseRefChildrenVisibilityOptions) {
+  // The visibility match up to the elements list. Visibility of elements is
+  // true if visible in the scroll container, false if outside.
+  const [childrenEls, setChildrenEls] = useState<HTMLElement[]>([]);
+  const [visibility, setVisibility] = useState<boolean[]>([]);
+
+  // Update list of children element
+  useEffect(
+    () => setChildrenEls(Array.from(ref.current?.children ?? []) as HTMLElement[]),
```

Comment:
Seems like this component won't work if the children change after first render, is that right?

Reply:
added back children